products/statement-generator/src/components/Input.tsx

```diff
@@ -101,16 +102,39 @@ const InputArea: React.FC<InputFieldProps> = ({
   adornment,
   className = '',
   shortWidth = false,
+  errorText,
 }) => {
   const utilityClasses = useUtilityStyles();
   const classes = useStyles({
     disabled,
     useShort: shortWidth || type === 'number',
   });
 
-  const [valid, isValid] = useState(false);
-  const checkValid = (e: string) => {
-    isValid(e.length > 0);
+  const [valid, isValid] = useState(true);
+
+  const [value, setValue] = useState(defaultValue || '');
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = e.currentTarget.value;
+
+    if (type === 'number') {
```

Comment:
This is very creative, but I think it might be a little much. I think we might be able to use the inherent ['min' attribute](https://www.w3schools.com/tags/att_input_min.asp) on the input to prevent negative values. Since our input component is utilizing the OutlinedInput component from MUI, check out the [inputProps](https://mui.com/material-ui/api/outlined-input/#OutlinedInput-prop-inputProps) to apply the min value. 

bonus points if you want to hide the down arrow when value == 0, but we can always have that as a future issue.

Reply:
Just made these changes, tried to go for the bonus points but i may have to open another issue for that as it might need some css styling.